### ISSUE TITLE
Buildscripts are not executed in a workspace

### ISSUE THREAD

**Issue by cole-h** [nmattia/naersk#64]
Right now, I'm attempting to build [Alacritty](https://github.com/alacritty/alacritty) using naersk, but the build always fails with `error: couldn't read /build/source/target/debug/build/alacritty-a1caefd600e8e902/out/gl_bindings.rs: No such file or directory (os error 2)`.

Alacritty has a few modules in its workspace: `alacritty`, which does the heavy-lifting of rendering, etc., and `alacritty_terminal`, which has most of the logic not directly tied to the display itself. To facilitate this, the `alacritty` submodule has a buildscript that generates GL bindings at compile-time. Using this project, however, these sub-`build.rs`s are not run resulting in a non-working binary being produced (when `singleStep = true`; otherwise it fails with the above error).

I tested this with a simple project with the following file structure and `default.nix`:

<details><summary>tree</summary>

```
.
├── build.rs
├── Cargo.lock
├── Cargo.toml
├── default.nix
├── mem1
│  ├── build.rs
│  ├── Cargo.toml
│  └── src
│     └── main.rs
├── mem2
│  ├── build.rs
│  ├── Cargo.toml
│  └── src
│     └── main.rs
└── nix
     ├── sources.json
     └── sources.nix
```
</details>
<details><summary>default.nix</summary>

```nix
let
    pkgs = import <nixpkgs> {};
    sources = import ./nix/sources.nix;
    naersk = pkgs.callPackage sources.naersk {};
in naersk.buildPackage ./.
```
</details>

The root `Cargo.toml` just consists of specifying the workspace members. `mem{1,2}` have the following in their `build.rs`:

```rust
fn main() {
    println!("cargo:warning=EXECUTED MEM1"); // or MEM2
}
```

and neither of these get executed.

Compare 

<details><summary>nix-build output</summary>

```
❯ nix-build --check
checking outputs of '/nix/store/xdfzx7d2rll4p4bkjppvmy7yk2m70vhb-rust-workspace-unknown.drv'...
unpacking sources
unpacking source archive /nix/store/0bjq3l2milnprg7znbjrj88cxqj9h224-buildtest
source root is buildtest
patching sources
configuring
[naersk] cargo_release: --release
[naersk] cargo_options: -Z unstable-options
[naersk] cargo_build_options: $cargo_release -j "$NIX_BUILD_CORES" --out-dir out --message-format=json-diagnostic-rendered-ansi
[naersk] cargo_test_options: $cargo_release -j "$NIX_BUILD_CORES"
[naersk] cargo_bins_jq_filter: select(.reason == "compiler-artifact" and .executable != null and .profile.test == false)
[naersk] cargo_build_output_json (created): /build/tmp.o9Y5MIAjH7
[naersk] pre-installing dep /nix/store/m9qkhpz0nq9abpkchz9k8vfd0rm6mg25-rust-workspace-deps-unknown
building
cargo -Z unstable-options build $cargo_release -j "$NIX_BUILD_CORES" --out-dir out --message-format=json-diagnostic-rendered-ansi
   Compiling mem1 v0.1.0 (/build/buildtest/mem1)
   Compiling mem2 v0.1.0 (/build/buildtest/mem2)
    Finished release [optimized] target(s) in 0.15s
running tests
cargo -Z unstable-options test $cargo_release -j "$NIX_BUILD_CORES"
   Compiling mem2 v0.1.0 (/build/buildtest/mem2)
   Compiling mem1 v0.1.0 (/build/buildtest/mem1)
    Finished release [optimized] target(s) in 0.19s
     Running target/release/deps/mem1-309da8c0a13658d5

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/release/deps/mem2-e75a6163211a8226

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

installing
[naersk] Using file /build/tmp.o9Y5MIAjH7 to retrieve build products
[naersk] found executable mem2 -> /nix/store/dr5p7czbwvavvqxjy2ws7bg9njliaj04-rust-workspace-unknown/bin/mem2
[naersk] found executable mem1 -> /nix/store/dr5p7czbwvavvqxjy2ws7bg9njliaj04-rust-workspace-unknown/bin/mem1
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/dr5p7czbwvavvqxjy2ws7bg9njliaj04-rust-workspace-unknown
shrinking /nix/store/dr5p7czbwvavvqxjy2ws7bg9njliaj04-rust-workspace-unknown/bin/mem2
shrinking /nix/store/dr5p7czbwvavvqxjy2ws7bg9njliaj04-rust-workspace-unknown/bin/mem1
strip is /nix/store/7bhi29ainf5rjrk7k7wyhndyskzyhsxh-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/dr5p7czbwvavvqxjy2ws7bg9njliaj04-rust-workspace-unknown/bin
patching script interpreter paths in /nix/store/dr5p7czbwvavvqxjy2ws7bg9njliaj04-rust-workspace-unknown
checking for references to /build/ in /nix/store/dr5p7czbwvavvqxjy2ws7bg9njliaj04-rust-workspace-unknown...
/nix/store/dr5p7czbwvavvqxjy2ws7bg9njliaj04-rust-workspace-unknown
```
</details>

to

<details><summary>cargo build output</summary>

```
❯ cargo build
   Compiling mem1 v0.1.0 (/tmp/buildtest/mem1)
   Compiling mem2 v0.1.0 (/tmp/buildtest/mem2)
warning: EXECUTED MEM1
warning: EXECUTED MEM2
    Finished dev [unoptimized + debuginfo] target(s) in 0.23s
```
</details>

----

TL;DR: Rust projects with a workspace setup containing member(s) with a `build.rs` file fail to run their buildscripts.

**Comment by nmattia** [comment]
@cole-h thanks for the nice repro, I'll give it a try tomorrow!
